src/addons/zwave-adapter/zwave-classifier.js

```diff
@@ -10,64 +10,158 @@
-const COMMAND_CLASS_SENSOR_BINARY  = 48; // 0x30
+const COMMAND_CLASS_SWITCH_BINARY = 37;       // 0x25
+const COMMAND_CLASS_SWITCH_MULTILEVEL = 38;   // 0x26
+const COMMAND_CLASS_SENSOR_BINARY  = 48;      // 0x30
+const COMMAND_CLASS_METER = 50;               // 0x32
 //const COMMAND_CLASS_SWITCH_ALL = 39;        // 0x27
-//const COMMAND_CLASS_CONFIGURATION = 112;    // 0x98
-
-const THING_TYPE_ON_OFF_SWITCH = 'onOffSwitch';
-const THING_TYPE_BINARY_SENSOR = 'binarySensor';
+const COMMAND_CLASS_CONFIGURATION = 112;    // 0x98
 
 class ZWaveClassifier {
 
   constructor() {
   }
 
   classify(node) {
-    for (var valueId in node.zwValues) {
-      var value = node.zwValues[valueId];
-
-      if (value.class_id == COMMAND_CLASS_SWITCH_BINARY) {
-        // This looks like an on/off switch
-        this.initOnOffSwitch(node, valueId);
-        return;
-      } else if (value.class_id == COMMAND_CLASS_SENSOR_BINARY) {
-        this.initBinarySensor(node, valueId);
-        return;
-      }
+    let binarySwitchValueId =
+      node.findValueId(COMMAND_CLASS_SWITCH_BINARY, 1, 0);
+    if (binarySwitchValueId) {
+      this.initSwitch(node, binarySwitchValueId);
+      return;
+    }
+
+    let binarySensorValueId =
+      node.findValueId(COMMAND_CLASS_SENSOR_BINARY, 1, 0);
+    if (binarySensorValueId) {
+      this.initBinarySensor(node, binarySensorValueId);
+      return;
     }
   }
 
-  initOnOffSwitch(node, valueId) {
-    node.type = THING_TYPE_ON_OFF_SWITCH;
-    node.properties.set('on',
-                        new ZWaveProperty(
-                          node,               // device
-                          'on',               // name
-                          {                   // property decscription
-                            type: 'boolean'
-                          },
-                          valueId             // valueId
-                        ));
+  addProperty(node, name, descr, valueId,
+              setZwValueFromValue, parseValueFromZwValue) {
+    let property = new ZWaveProperty(node, name, descr, valueId,
+                                     setZwValueFromValue,
+                                     parseValueFromZwValue);
+    node.properties.set(name, property);
+  }
+
+  initSwitch(node, binarySwitchValueId) {
+    node.type = Constants.THING_TYPE_ON_OFF_SWITCH;
+    this.addProperty(
+      node,                     // node
+      'on',                     // name
+      {                         // property decscription
+        type: 'boolean'
+      },
+      binarySwitchValueId       // valueId
+    );
+
+    const levelValueId =
+      node.findValueId(COMMAND_CLASS_SWITCH_MULTILEVEL, 1, 0);
+    if (levelValueId) {
+      node.type = Constants.THING_TYPE_MULTI_LEVEL_SWITCH;
+      this.addProperty(
+        node,                   // node
+        'level',                // name
+        {                       // property decscription
+          type: 'number',
+          unit: '%',
```

Comment:
Unit should be `'percent'` according to spec.

Reply:
The spec changed since I wrote that. It used to reference an ISO document which used %, but I'll change it since the updated spec now show 'percent'. Weird that the spec uses the singular version of the others since its almost never 1.